projects/feature-activation/0005-feature-activation-for-transactions.md

```diff
@@ -0,0 +1,182 @@
+
+# Reference-level explanation
+[Reference-level explanation]: #reference-level-explanation
+
+In this section, technical details are expanded for what was described above.
+
+## Reward Lock
+
+Before detailing FATX, let's describe how the RL works in general terms. Then, we'll be able to observe the analogy and define FATX. As explained before, there are two main contexts for calculating RL (new vertices are separated into txs and blocks).
+
+#### Dealing with new txs
+
+1. A tx is received, and it spends the reward of a block. It is verified for reward lock. Then, there are two possibilities:
+   1. If the current best block's height IS NOT enough to unlock the reward, the tx is invalid and is rejected.
+   2. If the current best block's height IS enough to unlock the reward, the tx is valid and remains in the mempool.
+
+#### Dealing with reorgs
+
+1. A reorg happens. Then, if the new best height is lower than the best height before the reorg,
+   1. Both txs that were already in the mempool, and confirmed txs that came back to the mempool, may have become invalid. For that, all txs in the mempool are re-verified (only for reward lock). If they're invalid, they're marked as such and removed from the mempool and the storage.
+
+#### Dealing with new blocks
+
+1. When a block is received, if one of the txs it confirms tries to spend a locked reward, the block is invalid and is rejected.
+
+This rule is only for guaranteeing no rewards can be spent too early. In practice, it's impossible for such tx to be in the mempool, as it would have been invalidated before by the previous rules. TODO: Is this correct? Why does this rule exist, is it for compatibility with the previous reward lock mechanism?
+
+## Feature Activation for Transactions
+
+Now, before describing the contexts above for FATX, let's make some definitions.
+
+### Premises
+
+1. After an `ACTIVE` block, all next blocks will also be `ACTIVE` (considering the same blockchain).
+2. Feature states for txs are only `ACTIVE` or `INACTIVE`.
+
+### Requirements
+
+Using the premises above, we must define a function that returns a state, given a transaction (note: we actually return multiple states for multiple features). That function must satisfy the following requirements:
+
+1. All txs that are received after (time-wise) an `ACTIVE` block in the best chain, must also be `ACTIVE`.
+2. When all txs are ordered by timestamp, there must not be an `INACTIVE` tx after an `ACTIVE` tx.
```

Comment:
Timestamp might be tampered with. You might say that "there must not be an INACTIVE tx after an ACTIVE tx when the transactions are topologically sorted".

Reply:
This is the only comment that may still be relevant in the new iteration. Tampering of timestamps is accounted for in the RFC, please read the session dedicated to this.